src/packagesystem.rs

```diff
@@ -40,9 +67,14 @@ fn rpm_parse_metadata(stdout: &[u8]) -> Result<ContentMetadata> {
         s.push_str(n);
         s
     });
+
+    // Map the version into Module struct
+    let mut modules_vec: Vec<Module> = pkgs.keys().map(|pkg_str| parse_evr(pkg_str)).collect();
```

Comment:
Not sure if should map the version str into Module struct when using `rpm -q`.

Reply:
I meant not sure if it is necessary to do the map, and it will compare using version if we do not have the Module struct. But agree to do the map.
Copy the change here:
```
        if let (Some(self_versions), Some(target_versions)) = (&self.versions, &target.versions) {
            compare_package_slices(self_versions, target_versions)
        } else {
            compare_package_versions(&self.version, &target.version)
        }
```